tests/test_decoders.py

```diff
@@ -79,6 +85,18 @@ def compress(body):
     assert response.read() == body
 
 
+@pytest.mark.parametrize("header_value", (b"deflate", b"gzip", b"br", b"identity"))
+def test_empty_content(header_value):
+    headers = [(b"Content-Encoding", header_value)]
+    response = httpx.Response(200, headers=headers, content=b"")
+    assert response.content == b""
+
+
+def test_decoders_empty_cases():
+    for decoder in {BrotliDecoder, DeflateDecoder, GZipDecoder, IdentityDecoder}:
+        assert decoder().decode(b"") == b""
```

Comment:
Let's change this to cover both `decode()` and `flush()`. So, for example...

```python
instance = decoder()
assert instance.decode(b"") == b""
assert instance.flush() == b""
```

Reply:
My two cents here: consider using parametrization to have one test *per decoder* instead of a single test. Makes it easy to know and debug which decoder is broken when it happens. :-)

```python
@pytest.mark.parametrize("decoder", [BrotliDecoder(), DeflateDecoder(), ...])
def test_decoders_empty_cases(decoder):
    assert decoder().decode(b"") == b""
    assert decoder.flush() == b""
```

Reply:
super you're right. I forgot. 
Is there any difference between `BrotliDecoder()` and `BrotliDecoder` using?

Reply:
> Is there any difference between BrotliDecoder() and BrotliDecoder using?

Depends on whether instanciating the decoder performs any kind of stateful stuff? If that is the case you can pass the class and create an `instance` as in Tom's snippet.

Reply:
After asked, I needed to add `MultiDecoder`. 
`instance = decoder([IdentityDecoder()]) if decoder == MultiDecoder else decoder()`